src/lib/pipeline/FetchUnit.cc

```diff
@@ -119,8 +122,12 @@ void FetchUnit::tick() {
 
     // If predecode fails, bail and wait for more data
     if (bytesRead == 0) {
-      assert(bufferedBytes_ < isa_.getMaxInstructionSize() &&
-             "unexpected predecode failure");
+      // TODO should be minimum?
+      // Assertion currently commented as bytesRead == 0 is used to force
+      // finishing tick on compressed instruction edge cases which are correct
+      // behaviour
+      //      assert(bufferedBytes_ < isa_.getMaxInstructionSize() &&
+      //             "unexpected predecode failure");
```

Comment:
Unsure whether this could cause problems with intentional bailing and "unexpected predecode failure" aliasing and triggering the same logic? @jj16791 any thoughts? Could return -1 for intentional bailing and handle separately

Reply:
Logic updated. Resolving as irrelevant